atef/widgets/config/utils.py

```diff
@@ -1212,3 +1223,121 @@ def addChild(self, child: TreeItem) -> None:
         child._row = len(self._children)
         self._children.append(child)
         self._columncount = max(child.columnCount(), self._columncount)
+
+
+class AddRowWidget(DesignerDisplay, QWidget):
+    """
+    A simple row widget with an add button.  To be used when space is precious
+    Connect a new-row slot to the add_button signal to create new rows
+    """
+    filename = 'add_row_widget.ui'
+
+    add_button: QtWidgets.QToolButton
+    row_label: QtWidgets.QLabel
+
+    def __init__(self, *args, text='Add new row', **kwargs):
+        super().__init__(*args, **kwargs)
+        self.add_button.setIcon(qta.icon('ri.add-circle-line'))
+        self.row_label.setText(text)
+
+
+class TableWidgetWithAddRow(QtWidgets.QTableWidget):
```

Comment:
Can we disable selection (set the mode to no selection) on these table widgets? It's not very useful here and it makes the rows harder to read by turning them to a darker blut:

![image](https://user-images.githubusercontent.com/10647860/232604882-3ad9c137-4caf-43f0-8aaf-539602d499f6.png)

https://doc.qt.io/qt-6/qabstractitemview.html#selectionMode-prop
https://doc.qt.io/qt-6/qabstractitemview.html#SelectionMode-enum

Reply:
addressed in [93849d0](https://github.com/pcdshub/atef/pull/159/commits/93849d0eee9f6b3d14c1a77d6be658f0e66cce93)

I had originally hooked the selection signal for updating the check row description updates, but ended up finding a different signal to do this